internal/host/hostutil/host_utils.go

```diff
@@ -298,3 +300,20 @@ func IsDiskEncryptionEnabledForRole(encryption models.DiskEncryption, role model
 		return false
 	}
 }
+
+func GetIgnitionEndpoint(cluster *common.Cluster, host *models.Host) (string, error) {
+	poolName := string(common.GetEffectiveRole(host))
+	if host.MachineConfigPoolName != "" {
+		poolName = host.MachineConfigPoolName
+	}
```

Comment:
Can we have a default "worker" here or in GetEffectiveRole?

Reply:
Not in GetEffectiveRole since the default of role is auto-assign

Reply:
Sure, just added it here. I hope it's a dead code, but we never know